### ISSUE TITLE
example-msg-generator: support bookmarking with bookmark(yes/no)

### ISSUE THREAD

**Issue by furiel** [syslog-ng/syslog-ng#3242]
`example-msg-generator` generates `num` number of messages after start. After reload, it generates `num` messages again. If reload/stop happens before the `num` messages are generated, the rest of the messages are forgotten.

For flow control tests, or queue related tests, it is desirable to have better control on the message generation.

A typical flow-control/queue related test looks like

- disconnect the destination.
- generate large number of messages, that fill the destination queue.
- reopen destination, with or without reload.
- expect that all messages arrive to the destination.

Your options are somewhat limited, because you either need to calculate the exact message number of messages that fill the queue (very hard), or use a source that bookmark aware (for example network).

If `example-msg-generator ` would support bookmarking, it could be a nice simplification for those tests.

This issue requests that `example-msg-generator` have a new `bookmark(yes/no)` option

- bookmark(yes)
`example-msg-generator` uses  a persist entry, where it stores the number of acknowledged messages. After reload, restart, it can continue the message generation from there.

- bookmark(no)
The current behavior. This one is also important, because `example-msg-generator` is a convenient development tool, that anyone can use to experiment with configuration iteratively. Productivity is further increased, if someone integrates a syslog-ng execution into their editor. For example: https://github.com/furiel/syslogngconf-mode

The default value should be `no`.

**Comment by gaborznagy** [comment]
> After reload, it generates num messages again. If reload/stop happens before the num messages are generated, the rest of the messages are forgotten.

This is only true if `num` option is set. Without this, messages generated endlessly within every 1s. What would happen in case of bookmark(yes)?

Or in general, what should be the bookmark that's persisted?

**Comment by furiel** [comment]
You are right. It all makes sense only when `num` is not null. We can detect this from grammar, and emit error or warning.

The bookmark could be the number of acknowledged messages.

**Comment by gaborznagy** [comment]
I did not mean that `bookmark()` option should not be used without `num()` option.
It can be used in that case as well, no need for warning messages.

I just wanted to point out a different scenario, when `num()` is not used.
Good feature idea, btw!
